wordpress-mobile/WordPress-Android
# React native/pass http error code

Updates to version of FluxC that returns full error object for failed requests. Takes that full response and extracts the http error code from it and passes that along to mobile gutenberg.

⚠️  Before merging this PR, the [associated FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1459) should be merged and the FluxC version reference here should be updated.

### Why We're Using a `Bundle` To Pass Error Information

This error information has to be passed from the `WordPress`module to the `WordPressEditor` module and then to the `react-native-gutenberg-bridge` module. In other words, `WordPress` depends on `WordPressEditor`, which depends on `react-native-gutenberg-bridge`. `WordpressEditor`, however, only has `react-native-gutenberg-bridge` as an ["implementation" dependency](https://github.com/wordpress-mobile/WordPress-Android/blob/fd1e8475e6898d906c6ae8887def97c11b821783/libs/editor/WordPressEditor/build.gradle#L82) (I'm guessing this was done for the very good reason of wanting to keep the core app clean of any React Native dependencies). As a result, `WordPress` does not know anything about `react-native-gutenberg-bridge` and `react-native-gutenberg-bridge` knows nothing about either `WordPress` or `WordPressEditor`. 

This means that if we made a class to hold the error data, no matter where we put it, one of either `WordPress` or `react-native-gutenberg-bridge` would not be able to use it. Instead, we would be forced to have one class in `WordPressEditor` for passing the error data from `WordPress` to `WordPressEditor`. Then, inside `WordPressEditor`, we would have to convert that first data class to a new data class from `react-native-gutenberg-bridge`, so that the information could be passed to that module (because `react-native-gutenberg-bridge` would not know about the class from `WordPressEditor`). 

Not only would we need these two, essentially identical, data classes, however. We would also need to convert the data class from `react-native-gutenberg-bridge` into a kind of object that can be sent over the javascript bridge to React Native (`WritableMap`). Fortunately, React Native has a helper method to automatically convert a `Bundle` into a `WritiableMap`. 

FWIW, one other option would be to use a `Map<String, Object>` instead of a `Bundle`. React Native also has a helper to convert that kind of object to a `WritableMap`. I did not go with that approach because that has even less type-safety than a `Bundle`.

If you think there's a better way to handle this, please let me know. It just seemed that using a well-typed data class with these errors would require a substantial amount of boilerplate for little, if any, gain since that data would then just have to be convered into a `WritableMap`, which is essentially the same thing as a `Bundle`.

### To Test

See test steps in https://github.com/wordpress-mobile/gutenberg-mobile/pull/1686

### PR submission checklist
